exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -16,19 +16,24 @@ import (
 // the producer are dropped.
 type boundedMemoryQueue[T any] struct {
 	component.StartFunc
+	QueueCapacityLimiter[T]
```

Comment:
Why embedding ?

Reply:
Same as above. Also, with changing `nopCapacityLimiter` to `requestsCapacityLimiter`, now we can reuse `Size()` as well